lib/eager_group/preloader.rb

```diff
@@ -22,37 +26,44 @@ def run
 
           @klass = @records.first.class
         end
-        record_ids = @records.map { |record| record.send(primary_key) }
-        unless definition = @klass.eager_group_definitions[definition_key]
-          next
+
+        Array.wrap(definition_key).each do |key|
```

Comment:
When eager group as this example.
`eager_group(posts: [:approved_comments_count, :comments_average_rating])`

definition_key is an array here, we actually need to run query twice.